clickhouse/tests/conftest.py

```diff
@@ -14,17 +15,46 @@
 @pytest.fixture(scope='session')
 def dd_environment():
     conditions = []
+
     for i in range(6):
         conditions.append(CheckEndpoints(['http://{}:{}'.format(common.HOST, common.HTTP_START_PORT + i)]))
         conditions.append(
             CheckDockerLogs(
                 'clickhouse-0{}'.format(i + 1), 'Logging errors to /var/log/clickhouse-server/clickhouse-server.err.log'
             )
         )
+
+    conditions.append(
+        ping_clickhouse(
+            common.CONFIG['server'],
+            common.CONFIG['port'],
+            common.CONFIG['username'],
+            common.CONFIG['password'],
+        )
+    )
+
     with docker_run(common.COMPOSE_FILE, conditions=conditions, sleep=10, attempts=2):
         yield common.CONFIG
 
 
 @pytest.fixture
 def instance():
     return deepcopy(common.CONFIG)
+
+
+def ping_clickhouse(host, port, username, password):
+    def _ping_clickhouse():
+        try:
+            client = clickhouse_driver.Client(
+                host=host,
+                port=port,
+                user=username,
+                password=password,
+            )
+            client.connection.connect()
+            client.connection.ping()
+            return True
+        except:
```

Comment:
Since we're importing `clickhouse_driver` anyway, should we be catching exceptions specific to its `Client`?

Reply:
alright

Reply:
However, the try/except is not needed since raising an exception will trigger a retry so we can get rid of it I think.

Reply:
oh, true!! Sweet, good catch!